openkongqi/records/sqlalch.py

```diff
@@ -38,40 +38,86 @@ def create_cnx(self, settings):
     def db_init(self):
         Base.metadata.create_all(bind=self.get_engine())
 
-    def is_duplicate(self, record):
-        dup_count = self._cnx.query(Record).filter_by(
-            ts=record[0].astimezone(pytz.utc),
-            uuid=record[1]
-        ).count()
-        return (not dup_count == 0)
-
     def get_engine(self):
         """Return engine url / data source name (DSN) of database."""
         return self._engine
 
-    def write_record(self, record, commit=True):
-        if not self.is_duplicate(record):
-            r = Record(ts=record[0].astimezone(pytz.utc),  # UTC timezne
-                       uuid=record[1],
-                       key=record[2],
-                       value=record[3])
-            self._cnx.add(r)
-        if commit:
+    def is_duplicate(self, record):
+        dup_count = self._cnx.query(Record).filter_by(
+            ts=record.ts, uuid=record.uuid, key=record.key
+        ).count()
+        return dup_count != 0
+
+    def write_records(self, data):
```

Comment:
Judging by the length of this function I can expect you to tell me to atomize it.

Here's what I think I can do:

Have a `write_record(uuid, record)` method where
* `uuid` is a uuid
* record is a dictionary with `ts` and `fields`

Have the complementary `write_records(uuid, records)` where:
* `uuid` is a uuid
* `records` is a list of dict

Now, due to the nature of the data scraped, we can get multiple UUIDs in one round, so I propose we actually have the main method to named `dump_data(data)` where:
* `data` will be a dictionary where key:pair == uuid:records

It won't take in UUIDs and it will just loop through each UUID and call `write_records`.